cluster/cluster.yaml

```diff
@@ -1857,6 +1857,61 @@ Resources:
             Resource: "*"
             Action:
             - "kms:Decrypt"
+{{- if and (index .Cluster.ConfigItems "spotio_account_id") (index .Cluster.ConfigItems "spotio_access_token") }}
+  SpotIOOceanCluster:
+    Type: Custom::ocean
+    Properties:
+      ServiceToken: !Sub "arn:aws:lambda:${AWS::Region}:178579023202:function:spotinst-cloudformation"
+      accessToken: "{{ .Cluster.ConfigItems.spotio_access_token }}"
+      accountId: "{{ .Cluster.ConfigItems.spotio_account_id }}"
+      autoTag: true
+      updatePolicy:
+        shouldUpdateTargetCapacity: false
+        shouldRoll: "false" # disable rolling update, this is managed by CLM,CLC
+        rollConfig:
+          roll:
+            batchSizePercentage: "25"
+      # API Docs: https://help.spot.io/spotinst-api/ocean/ocean-cloud-api/ocean-for-aws/create-2/
+      ocean:
+        name: "{{ .Cluster.Alias }}-{{ .Cluster.ID }}"
+        controllerClusterId: "{{ .Cluster.LocalID }}"
+        region: !Ref AWS::Region
+        autoScaler:
+          isEnabled: true
+          cooldown: 180 # 3 min. This seems to be the minimum value
+          # maximum resources which can be allocated to the cluster
+          # Use very high values as we have the limit set via the
+          # capacity.maximum (NodePool.MaxSize)
+          resourceLimits:
+            maxMemoryGib: 1500000
+            maxVCpu: 75000
+          down:
+            # evaluationPeriods: 3
+            maxScaleDownPercentage: 100
+          headroom: # disable headroom/preprovisioned instances
+            cpuPerUnit: 0
+            memoryPerUnit: 0
+            numOfUnits: 0
+          isAutoConfig: false
+        capacity:
+          minimum: 0
+          maximum: {{ nodeCIDRMaxNodes (parseInt64 .Cluster.ConfigItems.node_cidr_mask_size) (parseInt64 .Cluster.ConfigItems.reserved_nodes) }}
+          target: 0 # default target (should just be zero)
+        strategy:
+          fallbackToOd: {{ .Cluster.ConfigItems.spotio_ocean_fallback_to_ondemand }}
```

Comment:
Shouldn't this be per-node pool?

Reply:
This can only be set on the ocean and not per "virtual node group"

Reply:
Oh well. :(